src/linker/Linker.Dataflow/CustomAttributeSource.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Mono.Cecil;
+using System.Linq;
+
+namespace Mono.Linker.Dataflow
+{
+	class CustomAttributeSource
+	{
+		private readonly XmlFlowAnnotationSource[] _sources;
+
+		public CustomAttributeSource (LinkContext _context)
+		{
+			ArrayBuilder<XmlFlowAnnotationSource> annotationSources = new ArrayBuilder<XmlFlowAnnotationSource> ();
+			if (_context.AttributeDefinitions != null && _context.AttributeDefinitions.Count > 0) {
+				foreach (string a in _context.AttributeDefinitions) {
+					XmlFlowAnnotationSource xmlAnnotations = new XmlFlowAnnotationSource (_context);
+					xmlAnnotations.ParseXml (a);
+					annotationSources.Add (xmlAnnotations);
+				}
+			}
+			_sources = annotationSources.ToArray ();
```

Comment:
Why does it have to array and not list if it's not exposed?

Reply:
Now is using a List